manifests/component/expression_filter.pp

```diff
@@ -0,0 +1,26 @@
+define rsyslog::component::expression_filter (
+  Integer $priority,
+  String $target,
+  String $confdir,
+  Hash $conditionals,
+  Optional[String] $format = '<%= $content %>'
```

Comment:
Is this really ‘Optional’? If you don’t default to undef, in what situation can you set $format to undef? Since this is a defined type, you can’t even in hiera automatic data binding using ‘~’ (and I dislike that technically anyway!)

Reply:
fixed